includes/Core/Authentication/Authentication.php

```diff
@@ -1045,7 +1042,7 @@ private function get_authentication_oauth_error_notice() {
 	private function verify_proxy_setup_nonce() {
 		$nonce = $this->context->input()->filter( INPUT_GET, 'nonce', FILTER_SANITIZE_STRING );
 
-		if ( ! wp_verify_nonce( $nonce, Google_Proxy::ACTION_SETUP ) ) {
+		if ( ! $this->google_proxy->verify_nonce( $nonce ) ) {
```

Comment:
We might need to update this logic here:
* If the request comes from Site Kit, it is to redirect to the proxy. In that case, the nonce should be a regular WordPress nonce.
* Otherwise, the request comes from the proxy in which case it should be a "proxy nonce".

Handling these different requests in the same callback is arguably a bit confusing, we may wanna decouple them via a separate request. In that case, the proxy one should be kept intact. However, for now it might be okay to just try both? If `wp_verify_nonce` fails, try proxy nonce verification. With this we should be set up for both scenarios.